### ISSUE TITLE
Lignende ord

### ISSUE THREAD

**Issue by Thanerik** [Thanerik/Wign#17]
Muliggør i backend en link mellem forskellige ord, så tegn fra ordene fremstår i én samlet side. F.eks. [avocado](http://www.wign.dk/tegn/avocado) og [avokado](http://www.wign.dk/tegn/avokado).

**Comment by Ansty93** [comment]
https://www.quora.com/Is-there-an-online-synonym-API

Lidt spændende læsning omkring synonym API service. Måske noget man kan kigge på?

**Comment by Thanerik** [comment]
Tak, @Ansty93! Problemet er bare at der ikke nødvendigvis er ord med "alternativt" stavelse. Der er sikkert en masse ord som staves forskelligt, men har den nøjagtige samme betydning. F.eks. Ædelsten og Smykkesten. Hos Wikipedia kan man lave en redirect fra en ord til den anden - og der står under overskriften at man er blevet [omdirigeret fra XXX ord](https://da.wikipedia.org/wiki/Ædelsten).
I wikikoden kan man indstille en redirect, så ordet peger på en anden ord ([Kilde](https://da.wikipedia.org/w/index.php?title=%C3%A6delsten&action=edit)).

Hvordan løser vi det problem? Om vi skal sætte en ny felt i databasen med redirect, så hvis der eksisterer en værdi der, bliver man redirecteret og alle tegn fra den ord bliver "flyttet" med over til den nye ord? Ellers kan man lave en regexp sporing og tjekker om der eksisterer en ord med LIGNENDE formulering og giver en advarelse? Vi har vist en lignende funktionalitet i vores kode... unó momento!

**Comment by Thanerik** [comment]
Jeps! [Levenshtein distancen](https://en.wikipedia.org/wiki/Levenshtein_distance)! Jeg bruger funktionen til at vise nogle forslående ord hvis man har ledt efter en ord som ikke eksisterer - så bliver disse ord listet ([tegnController.php](https://github.com/Thanerik/Wign/blob/master/app/Http/Controllers/TegnController.php), line 42). Det kan bruges til at tjekke under oprettelsen af en tegn, om der er nogle ord med Levenshtein distance på maks 1. Men det er dog dyrt (performacemæssigt) at gennemgå hele databasen blot for at finde lignende ord...
